### PR TITLE
feat: add rules_py example using ipython shell

### PR DESCRIPTION
**Pull request by mattem**
Initial example of using rules_py. Planning to add a Django example as that has more moving parts, but this is simple case, and provides an interactive `ipython` shell.

### CODE REVIEW

File: rules_py/ipython/WORKSPACE [hunk 1]
```diff
@@ -0,0 +1,50 @@
+workspace(name = "rules_py_ipython_example")
```
Comment (alexeagle) [inline rules_py/ipython/WORKSPACE:1]: so far our convention in other Aspect rulesets has been that nested workspaces are under an e2e/ folder, and are just meant to protect us from bugs (minimal commenting and content)
User-facing examples are in the examples/ folder and rely on the root WORKSPACE file.

I know we've had a few different iterations on this. It's nice that you can just `bazel test ...` from the root here and cover all the examples

Comment (mattem) [reply rules_py/ipython/WORKSPACE:1]: Then write_source_files doesn't conform to this? 
https://github.com/aspect-build/bazel-examples/tree/main/write_source_files

> It's nice that you can just bazel test ... from the root here

"here" being this repo, or the rule set? As a lot of the examples in this repo have their own WORKSAPCE etc, so `bazel test ...` at the root of this repo would kinda do nothing.

File: rules_py/ipython/__main__.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+from IPython import start_ipython
+
+start_ipython()
```
[the remark this answers is not in the document]
Comment (mattem) [reply rules_py/ipython/__main__.py:3]: No

**Comment by mattem** [comment]
If ts gazelle is in rules_typescript, then this should be in the e2e folder too? https://github.com/aspect-build/bazel-examples/pull/4

Not trying to nit pick, I'm just not understanding the separation with what is going / already in here and the intended use.

**Comment by alexeagle** [comment]
let's chat about them - I'm sure we aren't consistent across rulesets. the bazel-examples repo is special since the whole thing is one big e2e - there are *only* nested WORKSPACEs

**Comment by mattem** [comment]
Reopened this in https://github.com/aspect-build/rules_py/pull/28

**Pull request closed without merging**